typescript/token/src/adapters/EvmTokenAdapter.ts

```diff
@@ -0,0 +1,213 @@
+import { BigNumber, PopulatedTransaction } from 'ethers';
+
+import {
+  BaseEvmAdapter,
+  ChainName,
+  MultiProtocolProvider,
+} from '@hyperlane-xyz/sdk';
+import {
+  Address,
+  Domain,
+  addressToByteHexString,
+  addressToBytes32,
+  bytes32ToAddress,
+  isValidAddressEvm,
+  strip0x,
+} from '@hyperlane-xyz/utils';
+
+import { MinimalTokenMetadata } from '../config';
+import {
+  ERC20,
+  ERC20__factory,
+  HypERC20,
+  HypERC20Collateral__factory,
+  HypERC20__factory,
+} from '../types';
+
+import {
+  IHypTokenAdapter,
+  ITokenAdapter,
+  TransferParams,
+  TransferRemoteParams,
+} from './ITokenAdapter';
+
+// Interacts with native currencies
+export class EvmNativeTokenAdapter
+  extends BaseEvmAdapter
+  implements ITokenAdapter
+{
+  async getBalance(address: Address): Promise<string> {
+    if (!isValidAddressEvm(address)) return '0';
+    const balance = await this.getProvider().getBalance(address);
+    return balance.toString();
```

Comment:
returning 0 if address isn't checksum seems bad

Reply:
I think this was initially done for the UI as a convenience to handle empty addresses while the form was being filled but I agree for this layer it's no longer a good idea. Will fix!